libraries/AP_ESC_Telem/AP_ESC_Telem.cpp

```diff
@@ -529,9 +528,32 @@ void AP_ESC_Telem::update()
                 _last_rpm_log_us[i] = _rpm_data[i].last_update_us;
             }
         }
+
+        if ((now_us - _rpm_data[i].last_update_us) > ESC_RPM_DATA_TIMEOUT_US) {
+            _rpm_data[i].data_valid = false;
+        }
     }
 }
 
+bool AP_ESC_Telem::rpm_data_within_timeout (const volatile AP_ESC_Telem_Backend::RpmData &instance, const uint32_t now_us, const uint32_t timeout_us) const
```

Comment:
could just pass in last_update_us and data_valid or make it a member of the backend

Reply:
So the backend doesn't actually have the RpmData struct. It's declared back there, but it actually lives and exists in the frontend, which is why I stuck the helper there. I thought about passing the instance flag, but found that half the places were iterating over a reference anyways. I can stick it in the RpmData struct, but wasn't sure that actually seemed easier.